app/src/main/java/com/hedvig/app/feature/genericauth/otpinput/SendOtpCodeUseCase.kt

```diff
@@ -0,0 +1,54 @@
+package com.hedvig.app.feature.genericauth.otpinput
+
+import androidx.annotation.StringRes
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.SendOtpCodeMutation
+import com.hedvig.app.R
+import com.hedvig.app.util.ErrorEvent
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+
+class SendOtpCodeUseCase(
+    private val apolloClient: ApolloClient
+) {
+    suspend operator fun invoke(otpId: String, otpCode: String): OtpResult {
+        return when (val result = apolloClient.mutate(SendOtpCodeMutation(otpId, otpCode)).safeQuery()) {
+            is QueryResult.Error -> OtpResult.NetworkError(result.message)
+            is QueryResult.Success -> parseSuccessResponse(result)
+        }
+    }
+
+    private fun parseSuccessResponse(result: QueryResult.Success<SendOtpCodeMutation.Data>): OtpResult {
+        val attempt = result.data.login_verifyOtpAttempt
+        return attempt.asVerifyOtpLoginAttemptError?.let {
+            val errorEvent = parseErrorCode(it.errorCode)
+            OtpResult.OtpError(errorEvent)
+        } ?: attempt.asVerifyOtpLoginAttemptSuccess?.let {
+            OtpResult.Success(it.accessToken)
+        } ?: OtpResult.OtpError(OtpErrorEvent.UNKNOWN)
+    }
+
+    private fun parseErrorCode(errorCode: String) = when (errorCode) {
+        "TOO_MANY_ATTEMPTS" -> OtpErrorEvent.TOO_MANY_ATTEMPTS
+        "ALREADY_COMPLETED" -> OtpErrorEvent.ALREADY_COMPLETED
+        "EXPIRED" -> OtpErrorEvent.EXPIRED
+        "WRONG_OTP" -> OtpErrorEvent.WRONG_OTP
+        else -> OtpErrorEvent.UNKNOWN
```

Comment:
Can we have these strigns as an Enum, or as a GraphQL type that the backend gives us or as anything typed really so that we don't forget to fix it whenever we change it?

Reply:
Yeah that would be ideal. I'll try and find a better solution for this.